### PR TITLE
Game Analyzer, BallModel, Ballpossession

### PR DESCRIPTION
**Pull request by mrlukasbos**
needs #205

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -300,7 +279,7 @@ set(INTERFACE_SOURCES
         ${PROJECT_SOURCE_DIR}/roboteam_ai/src/interface/TreeVisualizerWidget.cpp
         ${PROJECT_SOURCE_DIR}/roboteam_ai/src/interface/TreeVisualizerWidget.h
         ${PROJECT_SOURCE_DIR}/roboteam_ai/src/interface/RobotsWidget.cpp
-        ${PROJECT_SOURCE_DIR}/roboteam_ai/src/interface/RobotsWidget.h)
+        ${PROJECT_SOURCE_DIR}/roboteam_ai/src/interface/RobotsWidget.h roboteam_ai/src/analysis/GameAnalyzer.cpp roboteam_ai/src/analysis/GameAnalyzer.h roboteam_ai/src/analysis/AnalysisReport.cpp roboteam_ai/src/analysis/AnalysisReport.h roboteam_ai/src/analysis/RobotDanger.cpp roboteam_ai/src/analysis/RobotDanger.h)
```
Comment (mrlukasbos) [inline CMakeLists.txt]: `project source dir`

File: CMakeLists.txt [hunk 2]
```diff
@@ -147,7 +127,7 @@ set(UTILS_SOURCES
         ${PROJECT_SOURCE_DIR}/roboteam_ai/src/utilities/Hungarian.h
         ${PROJECT_SOURCE_DIR}/roboteam_ai/src/utilities/Pause.cpp
         ${PROJECT_SOURCE_DIR}/roboteam_ai/src/utilities/Pause.h
-        )
+        roboteam_ai/src/utilities/BallModel.cpp roboteam_ai/src/utilities/BallModel.h)
```
Comment (mrlukasbos) [inline CMakeLists.txt]: `project_source_dir`

**Comment by mrlukasbos** [comment]
The goal vision data is also incorrect, at least in the interface. Although I do suggest we merge this one soon because it's getting big

**Comment by rolfvdhulst** [comment]
> The goal vision data is also incorrect, at least in the interface. Although I do suggest we merge this one soon because it's getting big

What exactly do you mean with the goal vision data here? The merge blockades?

**Comment by mrlukasbos** [comment]
> > The goal vision data is also incorrect, at least in the interface. Although I do suggest we merge this one soon because it's getting big
> 
> What exactly do you mean with the goal vision data here? The merge blockades?

I'm not sure. the goalvision data in the interface is incorrect.

File: roboteam_ai/src/analysis/GameAnalyzer.cpp [hunk 3]
```diff
@@ -162,7 +162,7 @@ bool GameAnalyzer::isClosingInToGoal(roboteam_msgs::WorldRobot robot, bool ourTe
 void GameAnalyzer::start(int iterationsPerSecond) {
     if (!running) {
         ROS_INFO_STREAM_NAMED("GameAnalyzer", "Starting at " << iterationsPerSecond << " iterations per second");
-        auto delay = (unsigned) (1000/iterationsPerSecond);
+        auto delay = (unsigned) (1000/iterationsPerSecond); //TODO: Why is this integer division?
```
[the remark this answers is not in the document]
Comment (rolfvdhulst) [reply roboteam_ai/src/analysis/GameAnalyzer.cpp]: I found it in the gameanalyzer (which I didn't make here...)

**Pull request closed without merging**